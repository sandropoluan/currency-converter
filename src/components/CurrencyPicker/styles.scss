.currency-picker {
  padding: 8px;

  .add-currency-picker {
    font-weight: bold;
    color: teal;
    cursor: pointer;
  }

  .editor-picker {
    display: flex;
    flex-direction: row;

    :first-child{
      flex: 1;
      margin-right: 16px;
    }

    :last-child{
      margin-left: auto;
      width: 70px;
    }
  }
}
