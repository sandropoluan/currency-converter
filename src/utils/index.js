import React from "react";
import {Provider} from "react-redux";
import configureStore from "../redux/configureStore";


export const supportedCurrency = {
  'USD': 'United State Dollar',
  'CAD': 'Canadian Dollar',
  'IDR': 'Indonesian rupiah',
  'GBP': 'British Pound',
  'CHF': 'Swiss Franc',
  'SGD': 'Singapore Dollar',
  'INR': 'Indian Rupee',
  'MYR': 'Malaysian Ringgit',
  'JPY': 'Japanese Yen',
  'KRW': 'South Korean Won'
};

export const store = configureStore();

export const withReduxProvider = RenderedComponent => props => <Provider
  store={store}><RenderedComponent {...props}/></Provider>;
