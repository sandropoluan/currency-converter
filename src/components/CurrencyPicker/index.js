import React, {useState} from "react";
import {connect} from "react-redux";
import {addSelectedCurrency} from "../../redux/actions";
import "./styles.scss";

function CurrencyPicker(props) {
  const {listCurrency, addSelectedCurrency, selectedCurrencies} = props;

  const [editMode, setEditMode] = useState(false);
  const [selectedCurrency, changeCurrency] = useState('');

  const changeSelectedCurrency = e => {
    const {value} = e.target;

    if (!value) return;

    changeCurrency(value);
  };

  const submit = () => {
    if (!selectedCurrency) return;

    addSelectedCurrency(selectedCurrency);
    changeCurrency("");
    setEditMode(!editMode);
  };

  const toggleEditMode = () => {
    setEditMode(!editMode);
  };

  const availableOption = Object.keys(listCurrency).filter(item => selectedCurrencies.indexOf(item) === -1);

  return <div className="currency-picker">
    {!editMode && <div className="add-currency-picker" onClick={toggleEditMode}>+ Add more currencies</div>}
    {editMode && <div className="editor-picker">
      <select value={selectedCurrency} onChange={changeSelectedCurrency}>
        <option value="">--Please select--</option>
        {availableOption.map(item => <option key={item} value={item}>{item}</option>)}
      </select>
      <button className="btn btn-primary" onClick={submit}>Submit</button>
    </div>}
  </div>
}

export default connect(({app: {listCurrency, selectedCurrencies}}) => ({
  listCurrency,
  selectedCurrencies
}), {addSelectedCurrency})(CurrencyPicker);


