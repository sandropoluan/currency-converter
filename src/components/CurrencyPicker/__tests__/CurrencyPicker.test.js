import React from "react";
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { act, Simulate } from 'react-dom/test-utils';
import {withReduxProvider} from "../../../utils";
import CurrencyPicker from "../index.js";

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe(`Test CurrencyPicker`, () => {

  const TestComponent = withReduxProvider(CurrencyPicker);

  it('initial renders correctly', () => {
    const tree = renderer
      .create(<TestComponent/>);
    expect(tree).toMatchSnapshot();
  });


  it('initial renders correctly after button click', () => {

    const tree = renderer.create(<TestComponent/>);

    act(() => {
      ReactDOM.render(<TestComponent/>, container);
    });

    const button = container.querySelector('div.add-currency-picker');

    Simulate.click(button);

    expect(tree).toMatchSnapshot();

  });

});
