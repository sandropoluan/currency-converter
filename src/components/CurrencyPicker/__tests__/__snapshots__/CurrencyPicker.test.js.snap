// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CurrencyPicker initial renders correctly 1`] = `
<div
  className="currency-picker"
>
  <div
    className="add-currency-picker"
    onClick={[Function]}
  >
    + Add more currencies
  </div>
</div>
`;

exports[`Test CurrencyPicker initial renders correctly after button click 1`] = `
<div
  className="currency-picker"
>
  <div
    className="add-currency-picker"
    onClick={[Function]}
  >
    + Add more currencies
  </div>
</div>
`;
